### PR TITLE
thriftbp/thrifttest: Real example for NewBaseplateServer

### PR DESCRIPTION
**Pull request by fishy**
Convert the example of thrifttest.NewBaseplateServer to a real, full
file example.

### CODE REVIEW

File: thriftbp/thrifttest/server_example_test.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+func (srv mockedBaseplateService) IsHealthy(ctx context.Context, req *baseplatethrift.IsHealthyRequest) (r bool, err error) {
+	return !srv.Fail, srv.Err
+}
+
+// In real test this function needs to be named TestService instead,
+// but doing that will break this example.
+func ServiceTest(t *testing.T) {
+	// Initialize this properly in a real test
+	var store *secrets.Store
```
Comment (konradreiche) [inline thriftbp/thrifttest/server_example_test.go:25]: Should we reference here that this should be provided through `baseplate.New`?

File: thriftbp/thrifttest/server_example_test.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	success, err := client.IsHealthy(ctx, &baseplatethrift.IsHealthyRequest{
+		Probe: baseplatethrift.IsHealthyProbePtr(baseplatethrift.IsHealthyProbe_READINESS),
+	})
+
+	if err != nil {
+		t.Errorf("expected no error, got %v", err)
+	}
+
+	const expected = true
+	if success != expected {
```
Comment (konradreiche) [inline thriftbp/thrifttest/server_example_test.go:51]: Any thoughts on standardizing this to `got` & `want`?

**Review by konradreiche — APPROVED** [review]

File: thriftbp/thrifttest/server_example_test.go (hunk 1 above)
Comment (fishy) [reply thriftbp/thrifttest/server_example_test.go:26]: added mention of `secrets.NewTestSecrets` here.

**Review by konradreiche — APPROVED** [review]

File: thriftbp/thrifttest/server_example_test.go (hunk 2 above)
Comment (fishy) [reply thriftbp/thrifttest/server_example_test.go]: I personally usually use `expected` vs. `actual` but I guess `got` vs. `want` is shorter.

**Review by kylelemons — APPROVED** [review]

**Pull request merged**